### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by fjwillemsen**
I have finished building the homepage.
Please be so kind to review it.
It's mainly HTML and Sass, the most interesting part to review is probably the NavMenuService.
Thanks!

### CODE REVIEW

File: GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+@Component({
+    selector: 'app-circles-generated',
+    templateUrl: './circles.component.html',
+    styleUrls: ['./circles.component.scss']
+})
+export class CirclesGeneratorComponent implements AfterViewInit {
+    @ViewChild('CircleCanvas') canvasRef: ElementRef;
+    pageWidth: number;
+    pageHeight: number;
+    circleCoordinates: Array<[number, number]> = [];
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts]: Is the same as `[number,number][]`

File: GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+      const x = Math.random() * this.pageWidth;
+      const y = Math.random() * this.pageHeight;
+      if (this.distanceToCircles(x, y) === true) {
+        return [x, y];
+      }
+      return this.getCircleCoordinates(timesRan + 1);
+    }
+
+    distanceToCircles(x: number, y: number): boolean {
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts]: Not a good name. I expect by a method called distance that it should return number. You could also consider to make this method private.

File: GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    // Get random circle coordinates with a minimum distance from the others
+    getCircleCoordinates(timesRan: number): [number, number] {
+      if (timesRan > 10) {
+        throw Error('Infinite loop or can\'t find fitting coordinates');
+      }
+
+      const x = Math.random() * this.pageWidth;
+      const y = Math.random() * this.pageHeight;
+      if (this.distanceToCircles(x, y) === true) {
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts]: compare a boolean with true is unnecessary

File: GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    distanceToCircles(x: number, y: number): boolean {
+      let result = true;
+      const offset = this.max_size + this.min_distance;
+
+      this.circleCoordinates.forEach(coordinate => {
+        const diffX = coordinate['0'] - x;
+        const diffY = coordinate['1'] - y;
+
+        if (
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts]: If I'm right it was first `&&` then `||` so 

```
(diffX < offset || diffX > offset) ||
(diffY < offset || diffY > offset) &&
result === true
```
is the same as 
```
(diffX < offset || diffX > offset) ||
((diffY < offset || diffY > offset) && result === true)
```

this make that the second part only good be true as the first part is true and with a `||` the right path has not to be checked so is not needed

File: GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          (diffX < offset || diffX > offset) ||
+          (diffY < offset || diffY > offset) &&
+          result === true
+        ) {
+          result = true;
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/elements/circles/circles.component.ts]: also you get only the result of the last result from the itereation is that desirable?

So not you could rewrite the code to something like:

```typescript
private enoughDistanceToCircles(x: number, y: number): boolean {
      const offset = this.max_size + this.min_distance;

      for( let coordinate of this.circleCoordinates) {
        const diffX = coordinate['0'] - x;
        const diffY = coordinate['1'] - y;

        if (!(diffX < offset || diffX > offset))
          return false;
        }
      };

      return true;
    }
```

File: GoedeDoelenHelpen/ClientApp/src/app/home/home.component.ts [hunk 6]
```diff
@@ -1,10 +1,18 @@
...
   selector: 'app-home',
   templateUrl: './home.component.html',
-  styleUrls: ['./home.component.css']
+  styleUrls: ['./home.component.scss']
 })
 
-export class HomeComponent {
+export class HomeComponent implements OnInit {
+
+  constructor(private service: NavMenuService) {
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/home/home.component.ts]: mostly is the convention to name the service property just by its name. In the futere you possible need more than one service in a componenten so ```private navMenuService: NavMenuService```

File: GoedeDoelenHelpen/ClientApp/src/app/home/home.module.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
+// Modules
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/home/home.module.ts:1]: If you want a architecture with multiple modules you should also a module for shared services, modules and comonents

**Review by barld — CHANGES_REQUESTED** [review]

File: GoedeDoelenHelpen/ClientApp/src/app/nav-menu/nav-menu.service.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { Injectable, EventEmitter } from '@angular/core';
+
+@Injectable()
+export class NavMenuService {
+
+    private theme = 'default';
+
+    // Service to change the theme of the navmenubar
+    change: EventEmitter<string> = new EventEmitter();
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/nav-menu/nav-menu.service.ts]: https://stackoverflow.com/questions/36076700/what-is-the-proper-use-of-an-eventemitter#answer-36076701 mayby better to use a `subject` for this

Comment (fjwillemsen) [reply GoedeDoelenHelpen/ClientApp/src/app/nav-menu/nav-menu.service.ts]: This is from pre-release Angular 2 in 2016. We're at Angular 5 and it's 2018, hasn't been revoked yet. 
I will change it to a subject though.

File: GoedeDoelenHelpen/ClientApp/src/app/home/home.module.ts (hunk 7 above)
Comment (fjwillemsen) [reply GoedeDoelenHelpen/ClientApp/src/app/home/home.module.ts:1]: Let's talk about this friday, okay?

**Review by barld — APPROVED** [review]

**Pull request merged**